include/embedded_scheduler/scheduler.hpp

```diff
@@ -301,32 +301,41 @@ class EventLoop final
     template <typename ExecutionMonitor = std::monostate>
     [[nodiscard]] SpinResult<Clock> spin()
     {
-        auto next  = time_point::max();
-        auto now   = Clock::now();
-        auto worst = duration::zero();
+        SpinResult<Clock> result{.next_deadline  = time_point::max(),
+                                 .worst_lateness = duration::zero(),
+                                 .approx_now     = time_point::min()};
+        if (tree_.empty())
+        {
+            result.approx_now = Clock::now();
+            return result;
+        }
+
         while (auto* const evt = static_cast<EventProxy*>(tree_.min()))
         {
-            next = evt->getDeadline().value();  // The deadline is guaranteed to be set because it is in the tree.
-            auto lateness = now - next;         // Positive if late, zero if on-time.
-            if (lateness < duration::zero())    // Too early -- either we need to sleep or the time sample is obsolete.
+            // The deadline is guaranteed to be set because it is in the tree.
+            const auto deadline = evt->getDeadline().value();
+            if (result.approx_now < deadline)  // Too early -- either we need to sleep or the time sample is obsolete.
             {
-                now      = Clock::now();  // The number of calls to Clock::now() is minimized.
-                lateness = now - next;
-                if (lateness < duration::zero())  // Nope, even with the latest time sample we are still early -- exit.
+                result.approx_now = Clock::now();  // The number of calls to Clock::now() is minimized.
+                if (result.approx_now < deadline)  // Nope, even with the latest time sample we are still early -- exit.
```

Comment:
Note, that I also eliminated subtraction (which is over/under-flow prone strictly speaking). Now I subtract later, when I know for sure that `now >= deadline` (at line @ 333). Just in case, I will add separate unit test with deadline time near to `timepoint::max` - probably it will overflow even now...